management-service/services/segmenter_service.go

```diff
@@ -249,8 +249,10 @@ func (svc *segmenterService) ListSegmenters(
 			}
 			// UpdatedAt and CreatedAt fields are manually updated for custom segmenters but not global segmenters since
 			// global segmenters do not contain these fields
-			formattedCustomSegmenter.UpdatedAt = &customSegmenter.UpdatedAt
-			formattedCustomSegmenter.CreatedAt = &customSegmenter.CreatedAt
+			customSegmenterUpdatedAt := customSegmenter.UpdatedAt
+			customSegmenterCreatedAt := customSegmenter.CreatedAt
```

Comment:
Added these two lines because previously we were assigning the pointer of an iterating `customSegmenter`'s `UpdatedAt` and `CreatedAt` fields to the `customSegmenterUpdatedAt` and `customSegmenterCreatedAt` fields (these are to be returned to as responses of the API call) respectively. This resulted in all the custom segmenters returned to have the same `customSegmenterUpdatedAt` and `customSegmenterCreatedAt` values since they'd all have the same values as the last custom segmenter iterated in the for loop, which was the bug observed by @terryyylim.

![image](https://user-images.githubusercontent.com/36802364/179495757-568a95c0-20e9-4747-90eb-cf84c34c4d12.png)

The solution was to simply initialise new `time.Time` objects at each iteration and then set the  `customSegmenterUpdatedAt` and `customSegmenterCreatedAt` fields with pointers to these objects.

Reply:
Alternatively, it could be:
```golang
for idx, customSegmenter := range customSegmenters {
    ...
    formattedCustomSegmenter.UpdatedAt = &customSegmenters[idx].UpdatedAt
    ...
}
```

Reply:
Ah that looks wayyy cleaner 😅